### PR TITLE
Using new StringRouteConstraint for area constraint

### PR DESCRIPTION
**Pull request by iscifoni**
related to #4846 MapAreaRoute constraint not escaped 

@rynowak @Eilon @sebastienros

### CODE REVIEW

File: test/Microsoft.AspNetCore.Mvc.Core.Test/Builder/MvcAreaRouteBuilderExtensionsTest.cs [hunk 1]
```diff
@@ -33,7 +33,7 @@ public void MapAreaRoute_Simple()
                 kvp =>
                 {
                     Assert.Equal(kvp.Key, "area");
-                    Assert.IsType<RegexRouteConstraint>(kvp.Value);
+                    Assert.IsType<StringRouteConstraint>(kvp.Value);
```
Comment (pranavkm) [inline test/Microsoft.AspNetCore.Mvc.Core.Test/Builder/MvcAreaRouteBuilderExtensionsTest.cs]: Would it be useful to check if the value in `StringRouteConstraint` is the original value (and not escaped in any way)?

Comment (iscifoni) [inline test/Microsoft.AspNetCore.Mvc.Core.Test/Builder/MvcAreaRouteBuilderExtensionsTest.cs]: this check is present on StringRouteConstraint test in routing repository, i think at this point we do check if the right constraint is used.

i'll do it if you want

**Comment by iscifoni** [comment]
:up: :date:

File: test/Microsoft.AspNetCore.Mvc.Core.Test/Builder/MvcAreaRouteBuilderExtensionsTest.cs (hunk 1 above)
Comment (iscifoni) [inline test/Microsoft.AspNetCore.Mvc.Core.Test/Builder/MvcAreaRouteBuilderExtensionsTest.cs]: done :smile:

File: test/Microsoft.AspNetCore.Mvc.Core.Test/Builder/MvcAreaRouteBuilderExtensionsTest.cs [hunk 2]
```diff
@@ -9,6 +9,7 @@
 using Microsoft.Extensions.DependencyInjection;
 using Moq;
 using Xunit;
+using Microsoft.AspNetCore.Http;
```
Comment (Eilon) [inline test/Microsoft.AspNetCore.Mvc.Core.Test/Builder/MvcAreaRouteBuilderExtensionsTest.cs]: Sort 😄

File: test/Microsoft.AspNetCore.Mvc.Core.Test/Builder/MvcAreaRouteBuilderExtensionsTest.cs [hunk 3]
```diff
@@ -227,6 +228,48 @@ public void MapAreaRoute_DoesNotReplaceValuesForAreaIfAlreadyPresentInConstraint
                 });
         }
 
+        [Fact]
+        public void MapAreaRoute_DotCheck()
```
Comment (pranavkm) [inline test/Microsoft.AspNetCore.Mvc.Core.Test/Builder/MvcAreaRouteBuilderExtensionsTest.cs]: `MapAreaRoute_UsesPassedInAreaNameAsIs` (or something like that. (DotCheck doesn't necessarily say what this means).

**Comment by pranavkm** [comment]
LKGTM. I'll let @rynowak sign off on this.

File: test/Microsoft.AspNetCore.Mvc.Core.Test/Builder/MvcAreaRouteBuilderExtensionsTest.cs (hunk 2 above)
Comment (iscifoni) [inline test/Microsoft.AspNetCore.Mvc.Core.Test/Builder/MvcAreaRouteBuilderExtensionsTest.cs]: ops 😃

**Comment by iscifoni** [comment]
🆙 :date:

**Comment by rynowak** [comment]
:shipit: from me, I'll take care of merging

**Comment by rynowak** [comment]
Merged as  6e5187c1ae731b09ff2bd94d65fc6c670be735f3 - @iscifoni Thanks! :+1:

**Pull request closed without merging**